File: Backend/src/models/category_model.js
const { Schema, model } = require('mongoose');

const categorySchema = new Schema({
    title: { type: String, required: [true, "Title is required"], },
    description: { type: String, default: "" },
    image:{type:String,required:true},
    updatedOn: { type: Date },
    createdOn: { type: Date }
});

categorySchema.pre('save', function (next) {
    this.updatedOn = new Date();
    this.createdOn = new Date();

    next();
});

categorySchema.pre(['update', 'findOneAndUpdate', 'updateOne'], function (next) {
    const update = this.getUpdate();
    delete update._id;

    this.updatedOn = new Date();

    next();
});

const CategoryModel = model("Category", categorySchema);
module.exports = CategoryModel;

File: Frontend/lib/data/models/category/category_model..dart
class CategoryModel {
  String? sId;
  String? title;
  String? description;
  String? image;
  String? updatedOn;
  String? createdOn;

  CategoryModel(
      {this.sId,
        this.title,
        this.description,
        this.image,
        this.updatedOn,
        this.createdOn});

  CategoryModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    title = json['title'];
    description = json['description'];
    image = json['image'];
    updatedOn = json['updatedOn'];
    createdOn = json['createdOn'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['title'] = this.title;
    data['description'] = this.description;
    data['image'] = this.image;
    data['updatedOn'] = this.updatedOn;
    data['createdOn'] = this.createdOn;
    return data;
  }
}
